File: client/src/app/donor-pledge/pledge.ts
export interface Pledge {
  _id: string;
  comment: string;
  timeSlot: TimeSlot;
  name: string;
  amount: number;
  itemName: string;
}

export type TimeSlot = '' | 'Monday' | 'Tuesday' | 'Wednesday' | 'Thursday' | 'Friday';

File: server/src/main/java/umm3601/request/Pledge.java
package umm3601.request;

import org.mongojack.Id;
import org.mongojack.ObjectId;

@SuppressWarnings({"VisibilityModifier"})

public class Pledge {
  @ObjectId @Id

  @SuppressWarnings({"MemberName"})

  public String _id;

  public String name;
  public int amount;
  public String timeSlot;
  public String comment;
  public String itemId;

  @Override
  public boolean equals(Object obj) {
    if (!(obj instanceof Pledge)) {
      return false;
    }
    Pledge other = (Pledge) obj;
    return _id.equals((other._id));
  }

  @Override
  public int hashCode() {
    return _id.hashCode();
  }
}

